src/include/common/internal_types.h

```diff
@@ -1396,6 +1397,20 @@ typedef std::unordered_set<expression::AbstractExpression *,
                            expression::ExprHasher, expression::ExprEqualCmp>
     ExprSet;
 
+//===--------------------------------------------------------------------===//
+// Storage
+//===--------------------------------------------------------------------===//
+
+/* column_map_type used to store the layout of a tile_group
+ * <column offset> to <tile offset, tile column offset>
+ */
+typedef std::map<oid_t, std::pair<oid_t, oid_t>> column_map_type;
+
+/* tile_map_type used to store the mapping between a tile and its columns
+ * <tile index> to vector{<original column index, tile column offset>}
+ */
+typedef std::map<oid_t, std::vector<std::pair<oid_t, oid_t>>> tile_map_type;
```

Comment:
I'm not sold that these are common enough types for the whole system that they should be here ... but I guess it's okay.